src/prover_disk.hpp

```diff
@@ -119,21 +117,43 @@ class DiskProver {
         delete[] c2_buf;
     }
 
+    DiskProver(std::string filename, std::vector<uint8_t> memo, std::vector<uint8_t> id, uint8_t k,
+               std::vector<uint64_t> table_begin_pointers, std::vector<uint64_t> C2) :
+        filename(std::move(filename)),
+        memo(std::move(memo)),
+        id(std::move(id)),
+        k(k),
+        table_begin_pointers(std::move(table_begin_pointers)),
+        C2(std::move(C2))
+    {
+    }
+
+    DiskProver(DiskProver&& other) :
+        filename(std::move(other.filename)),
+        memo(std::move(other.memo)),
+        id(std::move(other.id)),
+        k(other.k),
+        table_begin_pointers(std::move(other.table_begin_pointers)),
+        C2(std::move(other.C2))
+    {
+    }
```

Comment:
It looks like you could just say = default here. Does your move constructor differ in some way? If so, that should probably be called out in a comment

Reply:
Default doesn't work here because `std::mutex` has the move constructor deleted.

Reply:
so, does the moved from object need to have its mutex locked? what does the mutex protect exactly? This sounds like it would need a bit more comprehensive description (in a comment)

Reply:
Good point, locked the mutex while moving but i think the move constructor itself does not really need a comment section about what the mutex does this should be part of the mutex definition imo and is a bit out of scope of this PR.

Reply:
What I meant is that; looking at that code, it looks like a mistake to not just do = default, and people may be tempted to make that change in the future. A comment would justify why it *isn't* just = default

Reply:
Ahh, i see, but if the mutex gets locked in the move constructor its pretty obvious without a comment why its not just default, isn't it? I'm going to wait for your reply in the outer scope (we should have only talked about it in this subsection haha) :) I can add that comment up without the lock.